src/llvm-remove-addrspaces.cpp

```diff
@@ -394,9 +394,21 @@ bool RemoveAddrspacesPass::runOnModule(Module &M)
         for (unsigned i = 0; i < Attrs.getNumAttrSets(); ++i) {
             for (Attribute::AttrKind TypedAttr :
                  {Attribute::ByVal, Attribute::StructRet, Attribute::ByRef}) {
-                if (Type *Ty = Attrs.getAttribute(i, TypedAttr).getValueAsType()) {
-                    Attrs = Attrs.replaceAttributeType(C, i, TypedAttr,
-                                                       TypeRemapper.remapType(Ty));
+                auto Attr = Attrs.
+#if JL_LLVM_VERSION >= 140000
+                    getAttributeAtIndex(
+#else
+                    getAttribute(
+#endif
```

Comment:
I think we generally prefer having two function calls here, instead of splitting it apart by an preprocessor directive.


```suggestion
#if JL_LLVM_VERSION >= 140000
                    getAttributeAtIndex(i, TypedAttr);
#else 
                   getAttribute( i, TypedAttr);
#endif
```

Reply:
Yep, I was going back and forth on that as well in terms of which one would be clearer.